### PR TITLE
Updates default connection and install deps

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR removes `oef` as default connection and replaces it with `stub`. This change is made to align with the new packages arrangement and to remove the `oef` as the focal point.

This PR also fixes dependencies and their installation across the repo.

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -17,6 +17,8 @@ deps =
 
 commands =
     pip install git+https://github.com/pytoolz/cytoolz.git#egg=cytoolz==0.10.1.dev0
+    pip install oef==0.8.1
+    pip install colorlog
```
Comment (marcofavorito) [inline tox.ini]: these can be defined in `deps`

File: tests/test_cli/test_create.py [hunk 2]
```diff
@@ -151,16 +151,16 @@ def test_connections_directory_exists(self):
...
+        """Check that the connections directory contains the stub directory."""
+        oef_connection_dirpath = Path(self.agent_name, "connections", "stub")
         assert oef_connection_dirpath.exists()
         assert oef_connection_dirpath.is_dir()
 
-    def test_oef_connection_directory_is_equal_to_library_oef_connection(self):
-        """Check that the oef connection directory is equal to the package's one (aea.connections.oef)."""
-        oef_connection_dirpath = Path(self.agent_name, "connections", "oef")
-        comparison = filecmp.dircmp(str(oef_connection_dirpath), str(Path(ROOT_DIR, "packages", "connections", "oef")))
+    def test_oef_connection_directory_is_equal_to_library_stub_connection(self):
```
Comment (marcofavorito) [inline tests/test_cli/test_create.py]: please change name

Comment (marcofavorito) [reply tests/test_cli/test_create.py]: also variable names where `oef` appears

File: tests/test_cli/test_create.py [hunk 3]
```diff
@@ -151,16 +151,16 @@ def test_connections_directory_exists(self):
         assert connections_dirpath.exists()
         assert connections_dirpath.is_dir()
 
-    def test_connections_contains_oef_connection(self):
-        """Check that the connections directory contains the oef directory."""
-        oef_connection_dirpath = Path(self.agent_name, "connections", "oef")
+    def test_connections_contains_stub_connection(self):
+        """Check that the connections directory contains the stub directory."""
+        oef_connection_dirpath = Path(self.agent_name, "connections", "stub")
```
Comment (marcofavorito) [inline tests/test_cli/test_create.py]: change name

File: docs/tac-skills.md [hunk 4]
```diff
@@ -40,7 +42,7 @@ You must set the start time to a point in the future `start_time: Nov 10 2019  1
 
 ### Run the TAC controller AEA
 ``` bash
-aea run
+aea run --connections oef
```
Comment (marcofavorito) [inline docs/tac-skills.md]: an alternative, completely optional: we can use `aea config set agent.default_connection oef` at some point earlier in the tutorial. The same for the other skills.

File: docs/tac-skills.md [hunk 5]
```diff
@@ -53,20 +55,24 @@ aea create tac_participant_two
 ### Add the tac participation skill to participant one
 ``` bash
 cd tac_participant_one
+aea add connection oef
 aea add skill tac_participation
 aea add skill tac_negotiation
+aea install
 ```
 
 ### Add the tac participation skill to participant two
```
Comment (marcofavorito) [inline docs/tac-skills.md]: in weather skill, the title is:
` ### Add the oef connection and the weather station skill `

here the `oef connection` part is missing. Would be nice to normalize the description of this step across skills tutorials.
Some ideas:
- remove the `add oef connection` part 
- change with `set the X skill up`

File: docs/tac-skills.md (hunk 4 above)
Comment (DavidMinarsch) [reply docs/tac-skills.md]: Yeah we could do that in another iteration. Will leave it for now.

File: docs/tac-skills.md (hunk 5 above)
Comment (DavidMinarsch) [reply docs/tac-skills.md]: Yeah we should refactor all guides once we have come to 0.2.0. For now I will leave it.

**Pull request merged**